forte/data/data_store.py

```diff
@@ -399,21 +400,40 @@ def _delete_entry_by_loc(self, type_id: int, index_id: int):
         # We then remove the entry data from the `type_id`th list.
         raise NotImplementedError
 
-    def get_entry(self, tid: int) -> Tuple[List, int, int]:
+    def get_entry(self, tid: int) -> Tuple[List, str, int]:
         r"""This function finds the entry with `tid`. It returns the entry,
         its `type_name`, and the index in the `type_name` list.
 
         Args:
             tid (int): Unique id of the entry.
 
         Returns:
-            The entry which `tid` corresponds to, its `type_id` and its index
+            The entry which `tid` corresponds to, its `type_name` and its index
             in the `type_name` list.
         """
+        if tid not in self.__entry_dict:
+            raise ValueError(f"Entry with tid {tid} not found.")
+        entry = self.__entry_dict[tid]
+        entry_type = entry[self.__entry_type_idx]
+        if entry_type not in self.__elements:
+            raise KeyError(f"Entry of type {entry_type} is not found.")
+
         # If the entry is an annotation, bisect the annotation sortedlist
         # to find the entry. May use LRU cache to optimize speed.
         # Otherwise, use `index_id` to find the index of the entry.
-        raise NotImplementedError
+        index_id = -1
+        if issubclass(get_class(entry_type), (Annotation, AudioAnnotation)):
+            entry_list = self.__elements[entry_type]
+            begin: int = entry_list.bisect_left(entry)
+            for i, e in enumerate(entry_list[begin:]):
+                if e[2] == entry[2]:
```

Comment:
We can make use of the [constant](https://github.com/asyml/forte/pull/635/files#diff-bed4b362357669e7f878c5c84fde9f9b6ed8f1a1d88b81a3e143a547ac8fa9c1R3) to replaced the hardcoded index here after #635 is merged.